### PR TITLE
Add margin documentation notebook

### LINKED ISSUE #227: Add margins section to the documentaiton.
Add content to https://lsdb.readthedocs.io/en/latest/tutorials/margins.html

Describe what margins are, how to generate, and what constraints are. Many plots, use resources from our demos.

### PR DESCRIPTION
**Pull request by smcguire-cmu**
Adds content to the `margins` section in the documentation closes #227

### CODE REVIEW

**Comment by smcguire-cmu** [comment]
There seem to be some issues building the notebook in the docs that I'm not seeing locally, so I'll need to dig into that more

File: docs/tutorials/margins.ipynb [hunk 1]
```diff
@@ -4,16 +4,377 @@
    "cell_type": "markdown",
```
Comment (nevencaplar) [inline docs/tutorials/margins.ipynb:4]: <p>explain more <code></code>right side of the catalog''</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/astronomy-commons/lsdb/pull/244/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

File: docs/tutorials/margins.ipynb [hunk 2]
```diff
@@ -82,16 +83,25 @@
     "collapsed": false
    },
    "source": [
-    "Then we load the object catalog, setting the `margin_cache` parameter to the margin catalog we have just loaded"
+    "Here we see the margin catalog that has been generated, in this case using a margin threshold of 10 arcseconds."
```
Comment (nevencaplar) [inline docs/tutorials/margins.ipynb:86]: Is there any way for the user to see it is 10 arcseconds? 

After this, it is ok to merge

Comment (smcguire-cmu) [reply docs/tutorials/margins.ipynb:86]: I tried printing the margin size from the catalog_info metadata in the output above, but I agree it's not the most obvious. Do you have any other ideas how to show it?

**Review by nevencaplar — APPROVED** [review]
Looks good to me

**Pull request merged**